src/messaging/ExchangeMgr.h

```diff
@@ -165,6 +166,8 @@ class DLL_EXPORT ExchangeManager : public SecureSessionMgrDelegate
 
     SecureSessionMgr * GetSessionMgr() const { return mSessionMgr; }
 
+    ReliableMessageManager * GetReliableMessageMgr() const { return const_cast<ReliableMessageManager *>(&mReliableMessageMgr); };
```

Comment:
This is never null, right?  Why not have it return a reference?

Also, the `const_cast` here seems like a const violation.  If we're really const, so should be the `ReliableMessageManager`; we can have a non-const getter returning non-const....

Reply:
Corrected. should return non-const pointer instead of const_cast since